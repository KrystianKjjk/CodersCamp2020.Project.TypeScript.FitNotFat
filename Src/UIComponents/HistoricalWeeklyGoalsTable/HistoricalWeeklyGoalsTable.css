.tile-title {
    text-transform: uppercase;
    line-height: 17px;
    letter-spacing: 0.035em;
    color: rgba(74, 72, 78, 0.7);
    padding-bottom: 10px;
    margin: 1rem 0;
    font-size: 1rem;
}

.tile-container {
    padding: 0px 20px 10px 20px;
    max-height: 35vh;
    overflow-y: scroll;
}
