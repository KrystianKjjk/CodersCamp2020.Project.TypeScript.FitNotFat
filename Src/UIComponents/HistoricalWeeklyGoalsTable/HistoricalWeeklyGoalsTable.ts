import {createTable, addRow} from "../ReusableTable/ReusableTable";
import generateTileComponent from "../TileComponent/TileComponent";

interface HistoricalWeeklyGoalsRow {
    date: string;
    weeklyGoal: string;
    startWeight: string;
    endWeight: string;
    achievedIn: string;
  }

function createHistoricalWeeklyGoalsTable(tableData: HistoricalWeeklyGoalsRow[]):HTMLDivElement{

    const container = document.createElement('div');
    container.className = 'tile-container';
    const title = document.createElement('p');
    title.className = 'tile-title';
    const titleContent = document.createTextNode('Historical weekly goals');
    title.appendChild(titleContent);
    container.appendChild(title);
    const tile=generateTileComponent(container);

    const table=createTable(['DATE','WEEKLY GOAL','START WEIGHT','END WEIGHT','ACHIEVED IN']);
    const addRowToHistoricalWeeklyGoalsTable = addRow(table);

    for(let i=0; i<tableData.length; i++){
    const row=tableData[i];
    const rowValues=Object.values(row);
    addRowToHistoricalWeeklyGoalsTable(rowValues);
    }
    container.appendChild(table);
    return tile;
}

export {createHistoricalWeeklyGoalsTable};

