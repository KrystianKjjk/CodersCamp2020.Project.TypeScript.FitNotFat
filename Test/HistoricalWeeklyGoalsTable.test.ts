import { WeeklyGoal } from '../Models/WeeklyGoal.model';
import {createHistoricalWeeklyGoalsTable} from '../Src/UIComponents/HistoricalWeeklyGoalsTable/HistoricalWeeklyGoalsTable'


describe('Tests for createHistoricalWeeklyGoalsTable',()=>{
    test('Should render only headers', ()=>{
       const table=createHistoricalWeeklyGoalsTable([]);
       expect(table).toMatchSnapshot();
    })

    test('Should render table with rows', ()=>{
        const table=createHistoricalWeeklyGoalsTable([{
            date: new Date(1995,11,17),
            weeklyGoal: WeeklyGoal.Gain,
            startWeight: 70,
            endWeight: 80,
            achieved: 'Yes',
          },
          {
            date: new Date(1994,12,18),
            weeklyGoal: WeeklyGoal.Lose,
            startWeight: 60,
            endWeight: 61,
            achieved: 'No',
          }]);
        expect(table).toMatchSnapshot();
    })
})