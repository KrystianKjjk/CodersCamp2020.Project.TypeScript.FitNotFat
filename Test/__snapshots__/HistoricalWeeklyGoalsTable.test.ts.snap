// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for createHistoricalWeeklyGoalsTable Should render only headers 1`] = `
<div
  class="tile"
>
  <div
    class="historical-weekly-goals-table tile-container"
  >
    <p
      class="tile-title"
    >
      Historical weekly goals
    </p>
    <table
      class="table-style"
    >
      <tr>
        <th>
          DATE
        </th>
        <th>
          WEEKLY GOAL
        </th>
        <th>
          START WEIGHT
        </th>
        <th>
          END WEIGHT
        </th>
        <th>
          ACHIEVED
        </th>
      </tr>
    </table>
  </div>
</div>
`;

exports[`Tests for createHistoricalWeeklyGoalsTable Should render table with rows 1`] = `
<div
  class="tile"
>
  <div
    class="historical-weekly-goals-table tile-container"
  >
    <p
      class="tile-title"
    >
      Historical weekly goals
    </p>
    <table
      class="table-style"
    >
      <tr>
        <th>
          DATE
        </th>
        <th>
          WEEKLY GOAL
        </th>
        <th>
          START WEIGHT
        </th>
        <th>
          END WEIGHT
        </th>
        <th>
          ACHIEVED
        </th>
      </tr>
      <tr>
        <td>
          17/12/1995
        </td>
        <td>
          Gain
        </td>
        <td>
          70
        </td>
        <td>
          80
        </td>
        <td>
          Yes
        </td>
      </tr>
      <tr>
        <td>
          18/01/1995
        </td>
        <td>
          Lose
        </td>
        <td>
          60
        </td>
        <td>
          61
        </td>
        <td>
          No
        </td>
      </tr>
    </table>
  </div>
</div>
`;
